keras/src/callbacks/orbax_checkpoint.py

```diff
@@ -195,23 +260,18 @@ def _save_checkpoint(self, step, logs=None):
         # names and structure)
         if self.save_weights_only:
             composite_state = {
+                "model_config": self.model.get_config(),
                 "trainable_variables": state_tree["trainable_variables"],
             }
             if "non_trainable_variables" in state_tree:
                 composite_state["non_trainable_variables"] = state_tree[
                     "non_trainable_variables"
                 ]
         else:
-            composite_state = state_tree
-
-        # --- Save Logic (V1 API) ---
-        # All processes participate in distributed checkpointing
-        # Checkpointer is configured to save unconditionally when
-        # save_pytree is called
-        if self.verbose > 0:
-            print_msg(
-                f"OrbaxCheckpoint: Triggering async save for step {step}..."
-            )
+            composite_state = {
+                "model_config": self.model.get_config(),
```

Comment:
Hmm... there is also the issue of the assets: https://github.com/keras-team/keras/blob/master/keras/src/saving/saving_lib.py

For this format to be fully self contained, it should also have the assets.